Sources/NDT7Settings.swift

```diff
@@ -11,73 +11,40 @@ import Foundation
 /// Settings needed for NDT7.
 /// Can be used with default values: NDT7Settings()
 public struct NDT7Settings {
-
-    /// URL for Web Socket.
-    public var url: NDT7URL
+    public var allServers: [NDT7ServerV2]
+    
+    public var currentServerIndex: Int
+    
+    /// Configured URL to run downlaod test against
+    public var downloadUrl: String
+    
+    /// Configured URL to run upload test against
+    public var uploadUrl: String
+    
+    /// Hostname of configured NDT7 server
+    public var hostname: String
 
     /// Timeouts
     public let timeout: NDT7Timeouts
 
     /// Skipt TLS certificate verification.
     public let skipTLSCertificateVerification: Bool
 
-    /// Use geo options to get a list of Mlab servers vs the closer one with false.
-    public let useGeoOptions: Bool
-
     /// Define all the headers needed for NDT7 request.
     public let headers: [String: String]
 
     /// Initialization.
-    public init(url: NDT7URL = NDT7URL(hostname: ""),
-                timeout: NDT7Timeouts = NDT7Timeouts(),
+    public init(timeout: NDT7Timeouts = NDT7Timeouts(),
                 skipTLSCertificateVerification: Bool = true,
-                useGeoOptions: Bool = false,
                 headers: [String: String] = [NDT7WebSocketConstants.Request.headerProtocolKey: NDT7WebSocketConstants.Request.headerProtocolValue]) {
-        self.url = url
         self.skipTLSCertificateVerification = skipTLSCertificateVerification
-        self.useGeoOptions = useGeoOptions
+        self.currentServerIndex = 0
```

Comment:
if you have an allServers array empty, the current server index should be optional and nil in that case.